code/drasil-gool/GOOL/Drasil/LanguageRenderer/CppRenderer.hs

```diff
@@ -1165,7 +1166,14 @@ instance MathConstant CppSrcCode where
     addMathHImport (mkStateVal double cppPi)
 
 instance VariableValue CppSrcCode where
-  valueOf = G.valueOf
+  valueOf vr' = do 
+    vr <- vr'
+    its <- zoom lensVStoMS getIter
+    let nvr = variableName vr 
+    if (nvr `elem` its) then (do
+        itvr <- iterator $ toState $ variableType vr
+        toState $ mkVal itvr (RC.variable vr)) 
+      else G.valueOf vr'
```

Comment:
This is what I was talking about with a do statement inside of a do statement, so make sure this is fine.

Reply:
You can use `mkStateVal` instead of `toState $ mkVal`. And `mkStateVal` takes a `VSType r`, so you won't need to use the do notation if you switch to that (i.e. you won't need to bind the iterator type to `itvr`).

Reply:
Such a construct is fine.